### PR TITLE
Use a common folder for the main rascal headers

### LINKED ISSUE #189: #include namespace for rascal headers
Currently, rascal headers are included directly, without any filesystem namespace. i.e. use `#include "math/math_utils.hh"` vs `#include "rascal/math/math_utils.hh"`.

This is not ideal for **users** of the C++ library who need to know which headers are from rascal, and which are not.

I think this needs to be updated. A nice way to do it is to separate headers and sources to look like:
```bash
librascal/
    include/
        rascal/
            math/
            representation/
    src/
        math/
        representation/
```

### PR DESCRIPTION
**Pull request by Luthaf**
Fix #189 

----

Tasks before review:

- [X] Add tests, examples, and complete documentation of any added functionality
    - [X] Make sure the autogenerated documentation appears in Sphinx and is
          formatted correctly (ask @max-veit if you need help with this task).
- [X] Run `make lint` on the project, ensure it passes
    - [X] Run `make pretty-cpp` and `make pretty-python`, check that the
          auto-formatting is sensible

### CODE REVIEW

File: examples/test_sph_harm_poles.cc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * file   test_sph_harm_poles.cc
+ * @file   examples/test_sph_harm_poles.cc
```
Comment (max-veit) [inline examples/test_sph_harm_poles.cc]: I thought we'd agreed that this file can be deleted. When did it get revived?

Comment (Luthaf) [reply examples/test_sph_harm_poles.cc]: It is deleted in #194

File: tests/test_adaptor_half_full_neighbour_list.cc [hunk 2]
```diff
@@ -25,10 +25,11 @@
...
-#include "structure_managers/adaptor_full_neighbour_list.hh"
-#include "structure_managers/adaptor_half_neighbour_list.hh"
 #include "test_structure.hh"
 
+#include "rascal/structure_managers/adaptor_full_neighbour_list.hh"
+#include "rascal/structure_managers/adaptor_half_neighbour_list.hh"
+
 #include <boost/test/unit_test.hpp>
 
 constexpr double TOLERANCE = 1e-14;
```
Comment (mastricker) [inline tests/test_adaptor_half_full_neighbour_list.cc]: This should probably go to the math tolerance, right? 
I'll make an issue.

File: tests/test_adaptor_increase_maxorder.cc [hunk 3]
```diff
@@ -26,9 +26,10 @@
...
  */
 
-#include "structure_managers/adaptor_half_neighbour_list.hh"
 #include "test_structure.hh"
 
+#include "rascal/structure_managers/adaptor_half_neighbour_list.hh"
+
 #include <boost/test/unit_test.hpp>
 
 constexpr double TOLERANCE = 1e-12;
```
Comment (mastricker) [inline tests/test_adaptor_increase_maxorder.cc]: This also.

**Pull request merged**